oauth2_provider/tests/test_authorization.py

```diff
@@ -0,0 +1,106 @@
+"""
+Tests various flows through using the Authorization Endpoint.
+"""
+from urlparse import urlparse, parse_qs
+
+import ddt
+from provider.oauth2.models import AccessToken
+import provider.scope
+
+from oauth2_provider.tests.base import OAuth2TestCase, IDTokenTestCase
+
+
+@ddt.ddt
+class ImplicitFlowTestCase(IDTokenTestCase):
+    """
+    Test OIDC-related extensions to the OAuth2 Authorization flows
+    implemented by this package.
+    """
+
+    def _get_auth_payload(self, location):
+        """
+        DRY helper.  Extract auth parameters from the redirect location as
+        returned during the implicit flow.
+        """
+        self.assertTrue(location.startswith(self.auth_client.redirect_uri + '#'))
+        return {k: v[0] for k, v in parse_qs(urlparse(location).fragment).items()}
+
+    def _get_access_token(self, scope):
+        """
+        DRY helper.  Find the access token generated during our auth request.
+        """
+        return AccessToken.objects.get(
+            user=self.user, client=self.auth_client, scope=provider.scope.to_int(*scope.split())
+        )
+
+    @ddt.data(
+        ('openid', 'id_token token'),
+        ('openid', 'id_token'),
+        ('openid profile', 'id_token'),
+        ('openid profile email', 'id_token'),
+    )
+    @ddt.unpack
```

Comment:
Do we support the other combinations that include `code`?  For example, the ones listed in:
http://openid.net/specs/oauth-v2-multiple-response-types-1_0.html#Combinations

If not, should we validate that they are not requested?